tests/harness.ps1

```
@@ -290,6 +290,15 @@ Write-Result "batch.req011.dircheck" "REQ-011: directory integrity check present
 $req012Patterns = @('HP_SKIP_ENTRY_SMOKE set; skipping entry-script smoke', 'HP_SKIP_EXE_SMOKERUN set; skipping EXE verification')
 $hasReq012 = ($req012Patterns | Where-Object { -not ($AllText -match $_) }).Count -eq 0
 Write-Result "batch.req012.skiphooks" "REQ-012: HP_SKIP_ENTRY_SMOKE and HP_SKIP_EXE_SMOKERUN execution-skip log lines present in run_setup.bat" $hasReq012 @{}
+# CLAUDE.md Active Backlog Item 45: :run_entry_smoke must gate on HP_PY actually existing
+# BEFORE :preflight_compile is ever called, so a failed env-create that fell through :die with
+# a broken HP_PY can never reach the PyInstaller build/warnfix/repair block. Scoped (not a
+# whole-file match) to the text between the :run_entry_smoke label and its first
+# "call :preflight_compile" -- both the guard's presence AND its ordering before the call are
+# proven together, since a guard placed anywhere after the call would fall outside this window.
+$item45Scope = [regex]::Match($AllText, '(?s):run_entry_smoke\r?\n.*?call :preflight_compile')
+$hasItem45Guard = $item45Scope.Success -and ($item45Scope.Value -match [regex]::Escape('if not exist "%HP_PY%" set "HP_NO_INTERPRETER=1"'))
+Write-Result "batch.entrysmoke.no_interpreter_guard" "CLAUDE.md Item 45: :run_entry_smoke gates HP_PY existence before :preflight_compile is called" $hasItem45Guard @{}
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Write-Result' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)